sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -442,6 +442,24 @@ public UnsafeRow copy() {
   }
 
   /**
+   * Copies this row, returning a UnsafeRow that stores its data to a user-provided byte[] buffer.
+   * User should ensure that provided buffer is large enough to store this row.
+   */
+  public UnsafeRow copyTo(byte[] buffer) {
```

Comment:
i was thinking about this ... have:

a static method fromByteArray(numBytes: Int) that is a factory method

and then add a 

copyFrom(UnsafeRow other)

which basically copies the data from other into the current buffer.

If the size is too small, change the buffer to expand it. 

Then it is easy to use. Just call copyFrom where we call copy right now.